docs/user/troubleshooting.md

```diff
@@ -165,4 +165,50 @@ You can either increase the capacity of your backend or reduce the volume of log
+* **Network Policy**: A Kubernetes `NetworkPolicy` is blocking egress traffic from the `telemetry-operator-gateway` pods to the OTLP endpoint.
+* **Backend Unavailability**: The backend service is crashing, restarting, or is otherwise unhealthy, causing it to drop active connections.
+
+### Solution
+
+1. **Verify the Endpoint**: Double-check the `endpoint` value in your pipeline's `output.otlp` section. Ensure the host and port are correct.
+2. **Verify TLS Configuration**:
+   * If your backend requires TLS, ensure `insecure` is not set to `true`.
+   * If your backend uses a custom Certificate Authority (CA), ensure the `ca` reference in your `tls` configuration is correct.
+   * If your backend requires mutual TLS (mTLS), verify that your `cert` and `key` references are correct and the secrets exist.
+3. **Check for Network Policies**: Look for `NetworkPolicy` resources in the `kyma-system` namespace that could be blocking egress traffic from the gateway pods. You may need to create a policy to explicitly allow this traffic.
+4. **Check Backend Health**: Ensure the OTLP backend service is running and accessible from within the cluster. You can test connectivity from the gateway pod using tools like `curl` (for HTTP) or `grpcurl` (for gRPC).
+
+## Transform or Filter Rules Not Working
+
+### Symptom
+
+You have configured a `transform` or `filter` section in your pipeline, but the data arriving at your backend is not modified, or data you expect to be dropped is still present.
+
+### Cause
+
+This usually happens for one of the following reasons:
+* **Incorrect Execution Order**: You are trying to filter data based on an attribute's original value, but a transformation rule has already modified it. Remember: transformations always run before filters.
+* **Incorrect Context Path**: Your OTTL expression is using the wrong path to an attribute. This is a common issue with filters, which require explicit context paths.
+* **Syntax Error**: Your OTTL expression has a syntax error. Because the default error mode is `ignore`, the processor logs the error and drops the invalid statement, so the pipeline continues to function without applying your rule.
+
+### Solution
+
+1. **Check the Gateway Logs**: Look for errors in the relevant gateway pod (`telemetry-metric-gateway`, `telemetry-trace-gateway`, etc.). Search for log entries containing `OTTL expression error`. These messages will point you to the exact syntax error in your statement. Execute: `kubectl logs -n kyma-system telemetry-metric-gateway | grep "OTTL expression error"`.
+2. **Verify the Execution Order**: Review your rules. If you have a transform that renames resource.attributes["foo"] to resource.attributes["bar"], your filter rule must check for bar, not foo.
+3. **Verify Context Paths in Filters**: Ensure your filter conditions use the full, explicit path to the attribute.
+   * **Incorrect**: `attributes["k8s.namespace.name"] == "default"`
```

Comment:
This incorrect expression should be rejected with a clear error message, so there should be no healthy pipeline. -> It is a different troubleshooting scenario as described